### ISSUE TITLE
CPU usage surges cause music stuttering

### ISSUE THREAD

**Issue by RohitAwate** [digimezzo/dopamine-windows#530]
The title is self explanatory. I tried to see if this is a hardware issue but Groove music runs just fine. Looked into the logs file and there's no error pertaining to this issue.

**Comment by Magentaize** [comment]
same here.

**Comment by Youxikong** [comment]
Same.

**Comment by DeusArcana** [comment]
Same here.
![cpu](https://user-images.githubusercontent.com/16895631/32580040-c0ace96e-c4a0-11e7-84df-b4cf063fe44b.png)

**Comment by Enlil12** [comment]
Same here, I have like 45gb of music, this happens to me when I hit shuffle on the entire library and after a couple of songs after the start of the reproduction I have a stuttering and I have to restart  the app.

**Comment by winger9** [comment]
Same here.
~5100 songs on shuffle. CPU usage is around 5% at the beginning and it increases after every song by 0,5-1%. The stuttering starts around 20%.

**Comment by darkon42** [comment]
Same here. ~7800 files, on shuffle. PC always on, dopamine always open but not always playing. It takes several days for the issue to appear (so minor annoyance).  Memory usage is fine, but CPU creeps up until stuttering occur.

**Comment by Proavtor** [comment]
So pretty application but the similar issue. I didn't face CPU overload but music often interrupted, making it impossible to use the application. Nothing unusual in log. I turned off all effects and improves but no avail. That is very critical bug which negates all other advantages unfortunately. =(

**Comment by 804260247** [comment]
Same here. The voice sounds stuttering.

**Comment by ttomovcik** [comment]
Test env:
- Library with around 40k songs
- Intel N4200 (4C/4T) @ 2,5GHz
- 8GB RAM
- Running on SSD
- Library loaded over network share, 10Gbit connection.

CPU usage (same on local and network):
- playing / MP3 (320kbps) - ~10%
- browsing library (first time): 30%
- searching for specific song: 70-100%
- FLAC playback: 40%.

After running Dopamine for about a hour normally, it started stuttering. When running in JIT Debugger, nothing.

**Comment by simongoricar** [comment]
Keeps happening to me as well after roughly an hour of use (~2500 tracks, as others mentioned - when the CPU usage reaches ~20%). 
For me, Dopamine uses anywhere from 0.1-0.8% of the CPU during minimized playback and around 10% when scrolling through the library.

Specs:
- Intel i5-5200U @ 2.2GHz
- 8GB RAM
- Tracks are loaded from local disk (HDD)
- Portable version @ 2.0 preview 13

**Comment by digimezzo** [comment]
@DefaultSimon , @ttomovcik , @804260247 , @Proavtor , @cjouny , @winger9 , @Enlil12 , @DeusArcana , @Magentaize , @Youxikong , @RohitAwate ,

Could you try out Dopamine 2.0 preview 14 which is available here? https://www.digimezzo.com/content/software/dopamine/

I'd like to know if that version still causes high CPU after 1 hours of more hours of use. Thanks!

**Comment by simongoricar** [comment]
I've been using *Preview 14* for a few weeks now and the problem definitely isn't gone yet. After long play sessions the CPU usage slowly climbs to around 20%, after which stuttering starts. In some cases however, the usage goes back down, only to climb up again later.

Another thing: there is an issue with sound stuttering when e.g. scrolling though the library or using other apps that cause momentary CPU spikes (alt-tabbing, closing a tab in Chrome, ...) *when the usage is getting close to 20%*. This doesn't happen when Dopamine's CPU usage is normal (~0.5-3%), so I'd say it is connected.

**Comment by ttomovcik** [comment]
@digimezzo Sorry for the delay.

tl;dr: it's better.

Memory usage is higher that usual, but that's fine. Started at 70-ish MB and after few minutes it settled down on 110.

And with the stuttering, sounds like it's gone. Depends on the target system / config. Using media controls on keyboard resulted in higher CPU usage from Dopamine, but nothing terrible. Also, the small popup that shows when next song starts playing might be what was causing the stuttering on my PC in the first place. Don't know why, but the first time it displayed, the music started stuttering a bit but then it went back to normal.

**Comment by Nord-Licht** [comment]
Same Problem here with Beta 3. I cannot say if it is better or worse than before, but when playing on shuffle (2832 Songs, 17GB, all MP3-VBR) the Sound begins to stutter. CPU usage is around 12%, GPU usage (?) is up to 2%. I tried some settings in hope it would get better, but without success (increasing latency setting, tried exclusive mode, increased process priority). The longer Dopamine is running, the worse the distortion gets. This is happening without any load otherwise.

System is a i7-4771 @3.5Ghz with 16Gigs of Ram, Storage is a Crucial MX500 SSD, Sound runs via Soundblaster Audigy Rx.

**Comment by digimezzo** [comment]
I'm trying to eliminate possible causes for this, hence the following question: do any of you have the "External control" setting enabled?

**Comment by simongoricar** [comment]
I left it on the default setting (disabled) and never touched it.

**Comment by Youxikong** [comment]
I turned it on.

**Comment by darkon42** [comment]
Never turned it on - left as default, I did not know the setting existed

**Comment by Nord-Licht** [comment]
"External control" setting is at default (off).

**Comment by agrmohit** [comment]
I turned 'external control' off and restarted Dopamine
The stutter I was experiencing is gone for now

**Comment by AKmatiAK** [comment]
External control disabled and problem exists. Multiple issues probably linked to this have been opened: #1360 #1338
